ospfd/ospf_vty.c

```diff
@@ -10013,63 +10012,29 @@ DEFPY(ospf_gr_helper_planned_only,
 }
 
 /* External Route Aggregation */
-DEFUN (ospf_external_route_aggregation,
-       ospf_external_route_aggregation_cmd,
-       "summary-address A.B.C.D/M [tag (1-4294967295)]",
-       "External summary address\n"
-       "Summary address prefix\n"
-       "Router tag \n"
-       "Router tag value\n")
+DEFPY(ospf_external_route_aggregation,
+      ospf_external_route_aggregation_cmd,
+      "[no] summary-address A.B.C.D/M$prefix [{tag (1-4294967295)$tag_val | metric (0-16777215)$m_val | metric-type (1-2)$m_type}]",
+      NO_STR
+      "External summary address\n"
+      "Summary address prefix (a.b.c.d/m)\n"
+      "Router tag\n"
+      "Router tag value\n"
+      "Metric\n"
+      "Advertised metric for this route\n"
+      "OSPF exterior metric type for summarised routes\n"
+      "Set OSPF External Type 1/2 metrics\n")
 {
 	VTY_DECLVAR_INSTANCE_CONTEXT(ospf, ospf);
 	struct prefix_ipv4 p;
-	int idx = 1;
-	route_tag_t tag = 0;
 	int ret = OSPF_SUCCESS;
 
-	str2prefix_ipv4(argv[idx]->arg, &p);
-
-	if (is_default_prefix4(&p)) {
-		vty_out(vty,
-			"Default address shouldn't be configured as summary address.\n");
-		return CMD_SUCCESS;
-	}
-
-	/* Apply mask for given prefix. */
-	apply_mask((struct prefix *)&p);
-
-	if (!is_valid_summary_addr(&p)) {
-		vty_out(vty, "Not a valid summary address.\n");
+	str2prefix_ipv4(prefix_str, &p);
```

Comment:
The `str2prefix_ipv4` transformation and the prefix validation are unnecessary since `DEFPY` checks that for you. Just use `prefix` instead.